python/ray/data/datasource/spark_datasource.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import os
+from typing import Iterator, List, Optional
+import numpy as np
+import pyarrow
+
+from ray.data.block import BlockMetadata
+from ray.data.datasource.datasource import Datasource, ReadTask
+from ray.util.annotations import PublicAPI
+
+
+logger = logging.getLogger(__name__)
+
+
+def check_requirements():
+    from ray.util.spark.utils import get_spark_session
+
+    spark = get_spark_session()
+    if spark.conf.get(
+        "spark.databricks.pyspark.dataFrameChunk.enabled", "false"
+    ).lower() != "true":
+        raise RuntimeError(
+            "In databricks runtime, if you want to use 'ray.data.from_spark' API, "
+            "you need to set spark cluster config "
+            "'spark.databricks.pyspark.dataFrameChunk.enabled' to 'true'."
+        )
+
+    if spark.conf.get(
+        "spark.databricks.acl.dfAclsEnabled", "false"
+    ).lower() != "false":
+        raise RuntimeError(
+            "In databricks runtime, if you want to use 'ray.data.from_spark' API, "
+            "you should use an assigned mode databricks cluster."
+        )
+
+
+def _persist_dataframe_as_chunks(spark_dataframe, bytes_per_chunk):
+    try:
+        from pyspark.databricks.sql.chunk import persistDataFrameAsChunks
+    except ImportError:
+        raise RuntimeError(
+            "Current Databricks Runtime version does not support Ray SparkDatasource."
+        )
+    return persistDataFrameAsChunks(spark_dataframe, bytes_per_chunk)
+
+
+def _read_chunk_fn(chunk_id_list) -> Iterator["pyarrow.Table"]:
```

Comment:
```suggestion
def _read_chunk_fn(chunk_ids) -> Iterator["pyarrow.Table"]:
```

using both `chunk_id_list` and `chunk_ids` seems confusing.